File: Ejercicio_03/Domain/Interfaces/Persona/IPersonaUseCases.cs
using Domain.DTOs;
using Domain.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Interfaces
{
    public interface IPersonaUseCases
    {
        List<Departamento> getListadoDepartamentos();
        PersonaWithNombreDepDTO getPersona(int id);
        List<PersonaWithNombreDepDTO> getPersonas();
        PersonaWithListadoDepDTO GetPersonaWithListadoDepDTO(int id);
        int updatePersona(int id, Persona persona);
        int deletePersona(int id);
        public int addPersona(Persona persona);
    }
}

File: Angular/CRUD/src/domain/interfaces/IPersonaUseCases.ts
import { PersonaDTO } from "../dtos/PersonaDTO";
import { Persona } from "../entities/Persona";

export interface IPersonaUseCases {
    getPersonas(): Array<PersonaDTO>
    getPersona(id: number): PersonaDTO
    updatePersona(id: number, persona: Persona): number
	addPersona(persona: Persona): number
	deletePersona(id: number): number
}